tests/filecheck/dialects/arith/canonicalize.mlir

```diff
@@ -3,15 +3,15 @@
 %lhsf32, %rhsf32 = "test.op"() : () -> (f32, f32)
 %lhsvec, %rhsvec = "test.op"() : () -> (vector<4xf32>, vector<4xf32>)
 
-%addf_0 = arith.addf %lhsf32, %rhsf32 : f32
+%addf = arith.addf %lhsf32, %rhsf32 : f32
 %addf_1 = arith.addf %lhsf32, %rhsf32 : f32
-%addf_vector_0 = arith.addf %lhsvec, %rhsvec : vector<4xf32>
+%addf_vector = arith.addf %lhsvec, %rhsvec : vector<4xf32>
 %addf_vector_1 = arith.addf %lhsvec, %rhsvec : vector<4xf32>
 
-"test.op"(%addf_0, %addf_vector_0) : (f32, vector<4xf32>) -> ()
+"test.op"(%addf, %addf_vector) : (f32, vector<4xf32>) -> ()
 
 // CHECK:        %lhsf32, %rhsf32 = "test.op"() : () -> (f32, f32)
 // CHECK-NEXT:   %lhsvec, %rhsvec = "test.op"() : () -> (vector<4xf32>, vector<4xf32>)
-// CHECK-NEXT:   %addf_0 = arith.addf %lhsf32, %rhsf32 : f32
-// CHECK-NEXT:   %addf_vector_0 = arith.addf %lhsvec, %rhsvec : vector<4xf32>
-// CHECK-NEXT:   "test.op"(%addf_0, %addf_vector_0) : (f32, vector<4xf32>) -> ()
+// CHECK-NEXT:   %addf = arith.addf %lhsf32, %rhsf32 : f32
```

Comment:
hm

Reply:
What do you mean on this one?

Reply:
That now,we do not start from zero if one occurrence is there.


I would expect that:

if one occurrence exists: addf
if multiple: addf_0, addf_1, addf_2

Reply:
As in, never printing the first one and directly starting at 0?
That would pretty costly, we would need to check ahead!
I'd also like to steer discussion: If we want to discuss better styles of doing it, can we maybe open an issue and discuss there? To keep this PR about the correct *behaviour*; I see it as relatively critical, as this currently is able to just parse valid IR and print it directly as invalid IR.

Reply:
But yeah, the choice of this fix is to just strip any meaning from `_[0-9]+` suffixes in value names. That's a super efficient one, both to ensure it's safe and at low cost.
Wasn't my first idea either, but I like it for those reasons.
Note that, you can still do fancy value naming, with anything other than numbers.

PS: Oh no. just remembered how much we like our numeric suffixes in DeVito :smiling_face_with_tear:. I would rather have a workaround there (We can add a suffix after the number  :shrug: ) and have xDSL core be correct before all